src/include/txn_inline.h

```diff
@@ -552,6 +568,11 @@ __wt_txn_visible_all(WT_SESSION_IMPL *session, uint64_t id, wt_timestamp_t times
     if (timestamp == WT_TS_NONE)
         return (true);
 
+    /* When reading a checkpoint, use the checkpoint state instead of the current state. */
+    if (WT_READING_CHECKPOINT(session))
```

Comment:
What if the checkpoint doesn't have an oldest timestamp?

Reply:
It will come through as 0 (see __meta_retrieve_timestamp in meta_ckpt.c) and then everything will be visible; in general that's what we want. For the specific case of the obsolete history store entries, it might not be what we want... but I'm not sure we can do anything better in that case. However, that should be limited to cases involving old checkpoints and upgrades. (If there was no oldest timestamp in the system when the checkpoint was taken, there can't really be any such thing as an obsolete history store entry...)